modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -105,6 +105,12 @@
   That is required eg. in `->lvalue :aggregation` call."
   0)
 
+(def ^:dynamic ^:private *next-alias-index*
+  "Tracks index of next alias for join compilation. It is bound in [[mbql->native]] to an atom valued 0. Hence every
+   compilation starts with a fresh 0. Indices are used in [[handle-join]] to make aliases unique. Index values
+   are gathered using `(swap *next-alias-index* inc)`, hence first used index is of value 1."
```

Comment:
It might make sense to create a function for  `(swap *next-alias-index* inc)`.

Reply:
Done.